connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -0,0 +1,156 @@
+import { ModelId } from "@dust-tt/types";
+import { CheerioCrawler, Configuration } from "crawlee";
+import turndown from "turndown";
+
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { upsertToDatasource } from "@connectors/lib/data_sources";
+import { Connector } from "@connectors/lib/models";
+import {
+  WebCrawlerConfiguration,
+  WebCrawlerFolder,
+} from "@connectors/lib/models/webcrawler";
+import {
+  reportInitialSyncProgress,
+  syncSucceeded,
+} from "@connectors/lib/sync_status";
+
+export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found.`);
+  }
+  const webCrawlerConfig = await WebCrawlerConfiguration.findOne({
+    where: {
+      connectorId,
+    },
+  });
+  if (!webCrawlerConfig) {
+    throw new Error(`Webcrawler configuration not found for connector.`);
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  let pageCount = 0;
+  let errorCount = 0;
+  const createdFolders = new Set<string>();
+
+  const crawler = new CheerioCrawler(
+    {
+      maxRequestsPerCrawl: 300,
+      maxConcurrency: 10,
+
+      async requestHandler({ $, request, enqueueLinks }) {
+        const extracted = new turndown()
+          .remove(["style", "script", "iframe"])
+          .turndown($.html());
+
+        const pageTitle = $("title").text();
+
+        await enqueueLinks();
+
+        const folders = getAllFoldersForUrl(request.url);
+        for (const folder of folders) {
+          if (!folder.startsWith(webCrawlerConfig.url)) {
+            continue;
+          }
+          if (createdFolders.has(folder)) {
+            continue;
+          }
+
+          const logicalParent =
+            folder === webCrawlerConfig.url ? null : getFolderForUrl(folder);
+          await WebCrawlerFolder.upsert({
+            url: folder,
+            parentUrl: logicalParent,
+            connectorId: connector.id,
+            webcrawlerConfigurationId: webCrawlerConfig.id,
+            ressourceType: "folder",
+            title: null,
+          });
+          createdFolders.add(folder);
+        }
+
+        const logicalParent =
+          request.url === webCrawlerConfig.url
+            ? null
+            : getFolderForUrl(request.url);
+        const updatedFolder = await WebCrawlerFolder.upsert({
+          url: request.url,
+          parentUrl: logicalParent,
+          connectorId: connector.id,
+          webcrawlerConfigurationId: webCrawlerConfig.id,
+          ressourceType: "file",
+          title: pageTitle,
+        });
+
+        await upsertToDatasource({
```

Comment:
Where do we enforce that we don't upsert pages that are not from the initial domain? We filter parents but do we filter documents? Isn't that configurable on CheerioCrawler as well?

Reply:
Domain is handled by the default settings of the crawler, and for folder I need to add it back, my initial code was not working properly so I am planning on adding a PR just for that.